### PR TITLE
Destination-snowflake: added host validation by pattern on UI

### LINKED ISSUE #23172: Snowflake destination should verify hostname
In this OC issue a user typed in an invalid hostname (included spaces), which caused oAuth to fail: https://github.com/airbytehq/oncall/issues/1259. If we want to prevent this in the future, we should add hostname validation to the UI. This is already possible in specs by adding pattern validation. Here is an example: 
https://github.com/airbytehq/airbyte/blob/b6c6b7affb67c29d16efd203b7a18c019c075c06/airbyte-integrations/connectors/source-google-pagespeed-insights/source_google_pagespeed_insights/spec.yaml#L26

### PR DESCRIPTION
**Pull request by etsybaev**
## What
If user typed in an invalid hostname (included spaces) - this will cause oAuth to fail: [airbytehq/oncall#1259](https://github.com/airbytehq/oncall/issues/1259). If we want to prevent this in the future, we should add hostname validation to the UI

## How
Added hostname validation to the UI
Selection_339

Selection_340

## Recommended reading order
1. `x.java`
2. `y.python`

## 🚨 User Impact 🚨
No breaking changes expected

## Pre-merge Checklist
Expand the relevant checklist and delete the others.

<details><summary><strong>New Connector</strong></summary>

### Community member or Airbyter

- [ ] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [ ] `docs/integrations/README.md`
    - [ ] `airbyte-integrations/builds.md`
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the connector is published, connector added to connector index as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

</details>

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub and connector version bumped by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

</details>

<details><summary><strong>Connector Generator</strong></summary>

- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)
- [ ] If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- [ ] The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- [ ] Documentation which references the generator is updated as needed

</details>

### CODE REVIEW

**Review by grishick — CHANGES_REQUESTED** [review]
need a stricter regex that matches IP address or hostname, but disallows protocol and query param parts of a URL

File: airbyte-integrations/connectors/destination-snowflake/src/main/resources/spec.json [hunk 1]
```diff
@@ -19,6 +19,7 @@
         ],
         "type": "string",
         "title": "Host",
+        "pattern": "^([^./?#]+\\.)?([^./?#]+\\.)?([^./?#]+\\.snowflakecomputing\\.com)$",
```
Comment (ryankfu) [inline airbyte-integrations/connectors/destination-snowflake/src/main/resources/spec.json:22]: Would like to add an `example` https://github.com/airbytehq/airbyte/blob/6f98a68e28c7847091229c2231a61a4bc4abe218/airbyte-integrations/connectors/source-amazon-seller-partner/integration_tests/spec.json#L66 so it's clear when someone types in the wrong host they have an idea of what the code is validating against

**Review by ryankfu — COMMENTED** [review]
Nit: on having an example since when a user inputs an incorrect host they will be presented with a regex which is not the best UX but also doesn't lead the user to self resolve this issue. An example would be a good way to ensure the user is able to understand what the host should look like for Snowflake

**Review by ryankfu — APPROVED** [review]
LGTM and thanks for clarifying examples already exist

**Comment by ryankfu** [comment]
cc: @airbytehq/frontend this fix adds a regex to validate hostnames for Snowflake, however, the issue we're seeing is that the error message is the `pattern` we're matching against and when an invalid hostname is presented it shows the regex. Is it possible to instead have it default to the `example` instead for a more human-readable format?

**Comment by lmossman** [comment]
@ryankfu we have actually just merged a feature that allows connector specs to declare a `pattern_descriptor` which serves this exact purpose: you can set a human-readable pattern string that should be displayed in the error as well as above the field. 

This feature is still behind a feature flag which is currently disabled so users won't see the change, but we plan to start doing some A/B testing with it over the next couple of weeks before enabling it for everyone. 

Here are the tentative docs that we will merge once this feature is enabled globally, check out the `Pattern descriptors` section: https://github.com/airbytehq/airbyte/pull/24125/files?short_path=2edd29e#diff-2edd29e95aa820bea04f7d721aa3d847d6a1068f236af0727bc87370a5f608e0

**Review by grishick — APPROVED** [review]

**Pull request merged**